### PR TITLE
Extract org name and repo name from url

### PR DESCRIPTION
**Pull request by SimonLab**
Update function to extract org and repo name from different
type of url, ssh, git, https

ref: #12

### CODE REVIEW

File: lib/helpers.ex [hunk 1]
```diff
@@ -58,14 +58,34 @@ defmodule Gitea.Helpers do
...
+    iex> Gitea.Helpers.get_org_repo_names("ssh://git@gitea-server.fly.dev:10022/theorg/myrepo.git")
+    {"theorg", "myrepo"}
+
+    iex> Gitea.Helpers.get_org_repo_names("https://gitea-server.fly.dev/myorg/public-repo.git")
+    {"myorg", "public-repo"}
+  """
   @spec get_org_repo_names(String.t()) :: {String.t(), String.t()}
-  defp get_org_repo_names(url) do
-    [org, repo] =
+  def get_org_repo_names(url) do
```
Comment (SimonLab) [inline lib/helpers.ex:74]: This function is not only used for `clone` but also in mock:
https://github.com/dwyl/gitea/blob/0562a5b750e0dd35cb0cc1e519f648c0ec78665d/lib/httpoison_mock.ex#L88

So I needed to make sure it works for git url but also for other http urls

**Comment by nelsonic** [comment]
When I run `mix c` on this branch with `mock: false` I get:

```sh
Finished in 24.6 seconds (24.6s async, 0.00s sync)
6 doctests, 29 tests, 0 failures

Randomized with seed 277879
----------------
COV    FILE                                        LINES RELEVANT   MISSED
100.0% lib/git_mock.ex                                55        7        0
100.0% lib/gitea.ex                                  244       48        0
100.0% lib/helpers.ex                                157       19        0
100.0% lib/http.ex                                   135       24        0
 96.2% lib/httpoison_mock.ex                         164       26        1
[TOTAL]  99.2%
----------------
Generating report...
Saved to: cover/
FAILED: Expected minimum coverage of 100%, got 99.2%.
```

Uncovered line is:

image

Just checked, the same line is not covered on `main` when `mock: false`

image

Let's fix it in a follow-up PR. 👍

**Comment by nelsonic** [comment]
Package published to https://hex.pm/packages/gitea/1.0.1 📦 🚀

**Comment by nelsonic** [comment]
Going to use this in demo app and see if the directory structure is created as we hope.
Then will move to integrate into the _actual_ app. 👍

**Pull request merged**